core/templates/dev/head/domain/exploration/SolutionObjectFactory.ts

```diff
@@ -17,108 +17,152 @@
  * domain objects.
  */
 
-require('domain/exploration/SubtitledHtmlObjectFactory.ts');
-require('domain/objects/FractionObjectFactory.ts');
-require('domain/objects/NumberWithUnitsObjectFactory.ts');
-require('filters/string-utility-filters/convert-to-plain-text.filter.ts');
-require('services/exploration-html-formatter.service.ts');
-require('services/html-escaper.service.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('SolutionObjectFactory', [
-  '$filter', 'ExplorationHtmlFormatterService', 'FractionObjectFactory',
-  'HtmlEscaperService', 'NumberWithUnitsObjectFactory',
-  'SubtitledHtmlObjectFactory',
-  function($filter, ExplorationHtmlFormatterService, FractionObjectFactory,
-      HtmlEscaperService, NumberWithUnitsObjectFactory,
-      SubtitledHtmlObjectFactory) {
-    var Solution = function(answerIsExclusive, correctAnswer, explanation) {
-      this.answerIsExclusive = answerIsExclusive;
-      this.correctAnswer = correctAnswer;
-      this.explanation = explanation;
-    };
+import { ConvertToPlainTextPipe } from
+  'filters/string-utility-filters/convert-to-plain-text.pipe';
+import { ExplorationHtmlFormatterService } from
+  'services/exploration-html-formatter.service';
+import { FractionObjectFactory } from 'domain/objects/FractionObjectFactory';
+import { HtmlEscaperService } from 'services/html-escaper.service';
+import { LoggerService } from 'services/contextual/logger.service';
+import { NumberWithUnitsObjectFactory } from
+  'domain/objects/NumberWithUnitsObjectFactory';
+import { SubtitledHtml, SubtitledHtmlObjectFactory } from
+  'domain/exploration/SubtitledHtmlObjectFactory';
+import { UnitsObjectFactory } from 'domain/objects/UnitsObjectFactory.ts';
 
-    Solution.prototype.toBackendDict = function() {
-      return {
-        answer_is_exclusive: this.answerIsExclusive,
-        correct_answer: this.correctAnswer,
-        explanation: this.explanation.toBackendDict()
-      };
-    };
+export interface Explanation {
```

Comment:
Maybe mention in name, this also corresponds to a backendDict?

Reply:
I'm sorry but I didn't get you. This is a interface declared for the explanation variable in backend dict -- so isn't this the right name? Please correct me if I'm wrong.

Reply:
I mean, you have backendDict written explicitly for the interface below, so just meant to rename this also  to `ExplanationBackendDict`, since I thought that was what this was supposed to be. Isn't it?

Reply:
Ok, I misunderstood before. Updated now!